phpBB/phpbb/db/migration/data/v330/storage_safe_filenames.php

```diff
@@ -0,0 +1,126 @@
+<?php
+/**
+*
+* This file is part of the phpBB Forum Software package.
+*
+* @copyright (c) phpBB Limited <https://www.phpbb.com>
+* @license GNU General Public License, version 2 (GPL-2.0)
+*
+* For full copyright and license information, please see
+* the docs/CREDITS.txt file.
+*
+*/
+
+namespace phpbb\db\migration\data\v330;
+
+class storage_safe_filenames extends \phpbb\db\migration\migration
+{
+	static public function depends_on()
+	{
+		return array(
+			'\phpbb\db\migration\data\v330\storage_track',
+		);
+	}
+
+	public function update_schema()
+	{
+		return array(
+			'add_columns'	=> array(
+				$this->table_prefix . 'storage'			=> array(
+					'safe_filename'		=> array('VCHAR', ''),
+				),
+			),
+		);
+	}
+
+	public function revert_schema()
+	{
+		return array(
+			'drop_columns'	=> array(
+				$this->table_prefix . 'storage'			=> array(
+					'safe_filename',
+				),
+			),
+		);
+	}
+
+	public function update_data()
+	{
+		return [
+			array('if', array(
+				($this->config['storage\\attachment\\provider'] == \phpbb\storage\provider\local::class),
```

Comment:
Is only a check, just in case.
For install this is not used, only for upgrades.

Reply:
Just in case what? Is it possible that a user has a different storage provider when running this? And what happens if they have? This config entry will be missing and I don't see a point why this should be the case.

Reply:
If he have been using this during the alpha and changed the storage provider yes.
Nothing happens, he will have an option in config that don't belong to this provider.